internal/pkg/workspace/workspace.go

```diff
@@ -160,6 +162,66 @@ func (ws *Workspace) ListWorkloads() ([]string, error) {
 	})
 }
 
+// Pipeline holds identifying information about a pipeline.
+type Pipeline struct {
+	Name string
+	Path string
+}
+
+// ListPipelines returns all pipelines in the workspace.
+func (ws *Workspace) ListPipelines() ([]Pipeline, error) {
+	var pipelines []Pipeline
+	// Look for legacy pipeline.
+	legacyPath, err := ws.PipelineManifestLegacyPath()
+	if err != nil {
+		return nil, err
+	}
+	manifest, err := ws.ReadPipelineManifest(legacyPath)
+	if err != nil {
+		log.Infof("Unable to read pipeline manifest at '%s'", legacyPath)
+	}
+	pipelines = append(pipelines, Pipeline{
+		Name: manifest.Name,
+		Path: legacyPath,
+	})
+	// Look for other pipelines.
+	pipelinesPath, err := ws.pipelinesDirPath()
+	if err != nil {
+		return nil, err
+	}
+	exists, err := ws.fsUtils.Exists(pipelinesPath)
+	if err != nil {
+		return nil, fmt.Errorf("check if pipeline manifest exists at %s: %w", pipelinesPath, err)
+	}
+	if !exists {
+		return pipelines, nil
+	}
+	// Look through the existent pipelines dir for pipeline manifests.
+	files, err := ws.fsUtils.ReadDir(pipelinesPath)
+	if err != nil {
+		return nil, fmt.Errorf("read directory %s: %w", pipelinesPath, err)
+	}
+	for _, file := range files {
+		// Ignore buildspecs.
+		if strings.HasSuffix(file.Name(), "buildspec.yml") {
+			continue
+		}
+		// Read manifests of moved legacy pipeline and any other pipelines.
+		if strings.HasSuffix(file.Name(), ".yml") {
+			path := filepath.Join(pipelinesPath, file.Name())
+			manifest, err := ws.ReadPipelineManifest(path)
+			if err != nil {
+				log.Infof("Unable to read pipeline manifest at '%s'", path)
+			}
+			pipelines = append(pipelines, Pipeline{
+				Name: manifest.Name,
+				Path: path,
+			})
```

Comment:
Do we need an else statement here? Otherwise we will keep appending Pipeline objects with an empty name when there is an error:
```suggestion
			if err != nil {
				ws.logger.Infof("Unable to read pipeline manifest at '%s'", path)
			} else {
			        pipelines = append(pipelines, Pipeline{
				       Name: manifest.Name,
				       Path: path,
			        })
			 }
```

Reply:
There is also a subtle change of moving the log to a `logger` field in the `Workspace` object that gets initialized in the constructor